test/test_simulation.c

```diff
@@ -0,0 +1,90 @@
+#include <unity.h>
+#include <stdio.h>
+#include "cJSON.h"
+#include "simulation_lifecycle/error.h"
+#include "simulation_lifecycle/utils/file.h"
+#include "simulation_lifecycle/simulation.h"
+
+void setUp(void) {
+    remove("data/simulation/output_config.json");
+}
+
+void tearDown(void) {
+    // clean stuff up here
+}
+
+void test_null_cjson(void) {
+    int res = build_simulation_scenario(NULL, NULL, NULL);
+    TEST_ASSERT_EQUAL(SIM_CONFIG_EMPTY, res);
+}
+
+void test_invalid_model_selection(void) {
+    cJSON *json = NULL;
+
+    read_json_file("./data/simulation/2_1_empty_config.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SELECTION_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/2_2_nonstring_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SELECTION_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/2_3_nonexisting_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SELECTION_INVALID, build_simulation_scenario(json, NULL, NULL));
+}
+
```

Comment:
Only your first test case is executed. I believe that you have to change all your paths to include: ../test/data/simulation

Reply:
Which is your working directory? I selected the `test` folder as the working directory when running the tests. Nonetheless, I'll check the paths as you suggested.